src/main/java/com/zaneta/movielibrary/controllers/MovieController.java

```diff
@@ -15,39 +19,73 @@ public class MovieController {
     MovieService movieService;
 
     @GetMapping("")
-    public List<Movie> getAll() {
-        return movieService.getAll();
+    public ResponseEntity getAll() {
+        ArrayList<Movie> result = new ArrayList<>(movieService.getAll());
+        if (result != null) {
+            return ResponseEntity.ok(result);
+        } else {
+            return ResponseConfig.getResponse(HttpStatus.NOT_FOUND, "list not found");
+        }
     }
 
     @GetMapping("/{id}")
-    public Movie getByID(@PathVariable("id") int id) {
-        return movieService.getById(id);
+    public ResponseEntity getByID(@PathVariable("id") int id) {
+        Movie result = movieService.getById(id);
+        if (result != null) {
+            return ResponseEntity.ok(result);
+        } else {
+            return ResponseConfig.getResponse(HttpStatus.NOT_FOUND, "movie not found");
+        }
     }
 
     @PostMapping("addmultiple")
-    public String add(@RequestBody List<Movie> movies) {
-        return movieService.add(movies);
+    public ResponseEntity add(@RequestBody List<Movie> movies) {
+        if (movies != null) {
+            return ResponseEntity.ok(movieService.add(movies));
+        } else {
+            return ResponseConfig.getResponse(HttpStatus.BAD_REQUEST, "movies were not added");
+        }
     }
 
     @PostMapping("add")
-    public String add(@RequestBody Movie movie) {
-        return movieService.add(movie);
+    public ResponseEntity add(@RequestBody Movie movie) {
+        if (movie != null) {
+            return ResponseEntity.ok(movieService.add(movie));
+        } else {
+            return ResponseConfig.getResponse(HttpStatus.BAD_REQUEST, " movie is not added");
+        }
     }
 
     @PutMapping("/{id}")
-    public int update(@PathVariable("id") int id, @RequestBody
+    public ResponseEntity update(@PathVariable("id") int id, @RequestBody
     Movie updateMovie) {
-        return movieService.updateMovie(id, updateMovie);
+        Movie movie = movieService.getById(id);
+        if (movie != null && movie.getName() != null && movie.getRating() != 0) {
```

Comment:
`movie.getRating() != 0` this mean that if rating is `null`, then the movie can be updated.
 
question: Can i have a movie with rating 0? does it mean that the movie is really bad?

I believe that checking if the rating is not null should be enough